src/Components/Server/src/Circuits/RemoteNavigationManager.cs

```diff
@@ -65,6 +72,31 @@ public void NotifyLocationChanged(string uri, string state, bool intercepted)
         NotifyLocationChanged(intercepted);
     }
 
+    public void HandleLocationChanging(int callId, string uri, bool intercepted)
```

Comment:
This should probably be async and have it await it on the CircuitHost calling code

Reply:
I believe this ties back into [this comment](https://github.com/dotnet/aspnetcore/pull/42638#discussion_r920372967), but let me know if I'm mistaken.

Reply:
@javiercn Ah, ok yes, I get it now. I've just updated everything - please let me know if it differs from what you had in mind.

/cc @SteveSandersonMS